src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,61 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_SIZE = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private int currentSize = INITIAL_SIZE;
+    private int size = 0;
+    private T[] table = createArray(currentSize);
 
+    private T[] createArray(int size) {
+        return (T[]) new Object[size];
+    }
```

Comment:
It would be better to just use constructor here:

```java
    @SuppressWarnings("unchecked")
    public MateStack() {
        this.stack = (T[]) new Object[DEFAULT_SIZE];
        currentSize = INITIAL_SIZE;
    }
```

Reply:
I'm sorry, I don't understand what exactly should I do here...

Reply:
It would be better practice to initialize  the table array not in the method createArray() but in constructor because:
Initialization Control: By initializing the array in the constructor, you have better control over the initialization process and can ensure that the array is always initialized properly when an instance of MateStack is created.

Avoiding Unnecessary Method: Initializing the array directly in the constructor avoids the need for a separate createArray method, which simplifies the code and makes it more straightforward.

Flexibility: Initializing the array in the constructor allows for more flexibility if you want to pass in different sizes or configurations when creating an instance of MateStack.